source/common/http/matching/inputs.cc

```diff
@@ -2,13 +2,47 @@
 
 #include "envoy/registry/registry.h"
 
+#include "source/common/http/utility.h"
+
 namespace Envoy {
 namespace Http {
 namespace Matching {
+
+Matcher::DataInputGetResult
+HttpRequestQueryParamsDataInput::get(const HttpMatchingData& data) const {
+  const auto maybe_headers = data.requestHeaders();
+
+  if (!maybe_headers) {
+    return {Matcher::DataInputGetResult::DataAvailability::NotAvailable, absl::nullopt};
+  }
+
+  const auto ret = maybe_headers->Path();
+  if (!ret) {
+    return {Matcher::DataInputGetResult::DataAvailability::NotAvailable, absl::nullopt};
+  }
+
+  Utility::QueryParams params;
+  if (url_decoded_) {
+    params = Http::Utility::parseAndDecodeQueryString(ret->value().getStringView());
+  } else {
+    params = Http::Utility::parseQueryString(ret->value().getStringView());
+  }
+  auto ItParam = params.find(query_param_);
+  if (ItParam == params.end()) {
+    return {Matcher::DataInputGetResult::DataAvailability::NotAvailable, absl::nullopt};
```

Comment:
I think the return code here doesn't match the expected behavior.
As you can see guide [here](https://github.com/envoyproxy/envoy/blob/abdbc449c61d43b6d0f9624d77bf57ffdb37d465/envoy/matcher/matcher.h#L201),  we should return {AllDataAvailable, absl::nullopt}

Reply:
Thanks for the link to the guide, this makes the semantic much clearer! Fixed here: https://github.com/envoyproxy/envoy/pull/24627/commits/5305a220ce918ae6fbac237d031bcb97fed2ff13#diff-f73e74c7f9395ff0a56e44f6fc5729a30576471e557bca2de38647b8af581e02R29